orthoinference/src/main/java/org/reactome/orthoinference/SkipTests.java

```diff
@@ -0,0 +1,219 @@
+package org.reactome.orthoinference;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+
+import org.gk.model.ClassAttributeFollowingInstruction;
+import org.gk.model.GKInstance;
+import org.gk.model.InstanceUtilities;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+import org.gk.schema.InvalidAttributeException;
+
+public class SkipTests {
+	
+	private static MySQLAdaptor dba;
+//	private static ArrayList<String> skipList = new ArrayList<String>();
+	static HashSet<String> skipList = new HashSet<String>();
+	
+	public static void setAdaptor(MySQLAdaptor dbAdaptor)
+	{
+		dba = dbAdaptor;
+	}
+	
+	// Skiplist was traditionally provided in a file, but since it's currently just 3 instances, I've just hard-coded them here.
+	public static void getSkipList(String skipListFilename) throws NumberFormatException, Exception
+	{
+		String[] pathwayIdsToSkip = {"162906","168254","977225"};
+		for (String pathwayId : pathwayIdsToSkip) 
+		{
+			GKInstance pathwayInst = dba.fetchInstance(Long.valueOf(pathwayId));
+			if (pathwayInst != null)
+			{
+				// Finds all ReactionLikeEvents associated with the skiplists Pathway and hasEvent attributes, and adds them to skiplist.
+				List<ClassAttributeFollowingInstruction> classesToFollow = new ArrayList<ClassAttributeFollowingInstruction>();
+				classesToFollow.add(new ClassAttributeFollowingInstruction(ReactomeJavaConstants.Pathway, new String[]{ReactomeJavaConstants.hasEvent}, new String[]{}));
+				String[] outClasses = new String[] {ReactomeJavaConstants.ReactionlikeEvent};
+				@SuppressWarnings("unchecked")
+				Collection<GKInstance> followedInstances = InstanceUtilities.followInstanceAttributes(pathwayInst, classesToFollow, outClasses);
+				
+				for (GKInstance entity : followedInstances)
+				{
+					skipList.add(entity.getDBID().toString());
+				}
+			}
+		}
+		// Generates new skiplist file 
```

Comment:
The code below looks like it's _reading_ a file, not generating one.

Reply:
I'm also confused: a comment earlier on says the instances to be skipped will be hard-coded, but then it seems to read more instances to skip from a file?

Reply:
Sounds like comments from different stages of me working on it. The skip list portion of the code was completed in the summer and has worked comparatively to the Perl version of the code since then. I'll clean up the comments